utils/zerovec/src/zerovec/mod.rs

```diff
@@ -392,6 +392,58 @@ where
         self.as_slice().iter().map(T::from_unaligned)
     }
 
+    /// Gets a mutable iterator over the elements. If you just need to mutate
+    /// elements in-place, consider using [`ZeroVec::for_each_mut()`] instead.
+    ///
+    /// This will convert the ZeroVec into an owned ZeroVec if not already the case.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use zerovec::ZeroVec;
+    /// use zerovec::ule::AsULE;
+    ///
+    /// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
+    /// let mut zerovec: ZeroVec<u16> = ZeroVec::parse_byte_slice(bytes).expect("infallible");
+    ///
+    /// for i in zerovec.iter_mut() {
+    ///     *i = (u16::from_unaligned(i) + 1).into();
+    /// }
+    ///
+    /// assert_eq!(zerovec.to_vec(), &[212, 282, 422, 462]);
+    /// ```
+    #[inline]
+    pub fn iter_mut(&mut self) -> impl Iterator<Item = &mut T::ULE> + '_ {
+        self.make_mut().iter_mut()
+    }
+
+    /// Mutates each element according to a given function, meant to be
+    /// a more convenient version of [`ZeroVec::iter_mut()`] which serves fewer use cases.
+    ///
+    /// This will convert the ZeroVec into an owned ZeroVec if not already the case.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use zerovec::ZeroVec;
+    /// use zerovec::ule::AsULE;
+    ///
+    /// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
+    /// let mut zerovec: ZeroVec<u16> = ZeroVec::parse_byte_slice(bytes).expect("infallible");
+    ///
+    /// zerovec.for_each_mut(|item| *item += 1);
+    ///
+    /// assert_eq!(zerovec.to_vec(), &[212, 282, 422, 462]);
+    /// ```
+    #[inline]
+    pub fn for_each_mut(&mut self, mut f: impl FnMut(&mut T)) {
+        self.make_mut().iter_mut().for_each(|item| {
+            let mut aligned = T::from_unaligned(item);
+            f(&mut aligned);
+            *item = aligned.as_unaligned()
```

Comment:
`question (n-b)`: would perf be better if we did:

```rust
if f(&mut aligned) {
  *item = aligned.as_unaligned()
}
```

my reasoning - in scenarios I encountered, I want to iterate over the vec, but I only modify a small subset - usually just one field. So while I need to `from_unaligned` each one to find the right one, we may be able to save on `as_unaligned` for most items.

Reply:
I think due to inlining it should be able to avoid that cost anyway. I'd rather not microoptimize in ways that impact the API